pkg/registration/spoke/options.go

```diff
@@ -38,12 +51,17 @@ func NewSpokeAgentOptions() *SpokeAgentOptions {
 		clientCertHealthChecker: &clientCertHealthChecker{
 			interval: ClientCertHealthCheckInterval,
 		},
+		HubConnectionTimeoutSeconds: 600, // by default, the timeout is 10 minutes
 	}
 
 	options.bootstrapKubeconfigHealthChecker = &bootstrapKubeconfigHealthChecker{
 		bootstrapKubeconfigSecretName: &options.BootstrapKubeconfigSecret,
 	}
 
+	options.reSelectChecker = &reSelectChecker{
```

Comment:
no need to declare separately, just put this under line 54

Reply:
Updated.